src/lib/y2network/autoinst_profile/dns_section.rb

```diff
@@ -103,7 +103,7 @@ def init_from_hashes(hash)
       #
       # @return [Boolean] Result true on success or false otherwise
       def init_from_network(dns, hostname)
-        @dhcp_hostname = hostname.dhcp_hostname
+        @dhcp_hostname = hostname.dhcp_hostname == :any
```

Comment:
This is a fix in the `dns_section` that probably should be backported to SLE-15-SP2